internal/expr/expr_test.go

```diff
@@ -570,17 +570,17 @@ func (s *ExprSuite) TestMapError(c *C) {
 		"all output into map star",
 		"SELECT &M.* FROM person WHERE name = 'Fred'",
 		[]any{sqlair.M{}},
-		"cannot prepare statement: output expression: columns must be specified for map with star: &M.*",
+		"cannot prepare statement: output expression: cannot generate columns for non-struct type: &M.*",
```

Comment:
I preferred the error message that we had in the past, it clearly says how to fix they issue (i.e. by specifying the columns). I would keep the new `non-struct type`  though.